### PR TITLE
Add a bpf auth flush command

### PR DESCRIPTION
**Pull request by meyskens**
This command in cilium-agent allows the user to flush the auth map. This can be used to force re-auths between all entities for debugging purposes.

This is useful to force an auth handshake on a given agent node when debugging. (once https://github.com/cilium/cilium/issues/27183 is fixed)

This function works similar to the `nat flush` command. 

```release-note
Add `cilium bpf auth flush` command for debugging purposes
```

### CODE REVIEW

File: cilium/cmd/bpf_auth_flush.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+			Fatalf("Cannot load auth bpf map: %s", err)
+		}
+
+		entries := 0
+		deleteEntry := func(key *authmap.AuthKey, _ *authmap.AuthInfo) {
+			if key == nil {
+				return
+			}
+			authMap.Delete(*key)
```
Comment (sayboras) [inline cilium/cmd/bpf_auth_flush.go]: probably checking returned err here, and print out if not nil. Also, only do entries++ if there is no error.

**Comment by sayboras** [comment]
Also you might need to run `make -C Documentation update-cmdref` to generate related docs

File: cilium/cmd/bpf_auth_flush.go (hunk 1 above)
Comment (meyskens) [reply cilium/cmd/bpf_auth_flush.go]: good point!

File: cilium/cmd/bpf_auth_flush.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+			Fatalf("Cannot load auth bpf map: %s", err)
+		}
+
+		entries := 0
+		deleteEntry := func(key *authmap.AuthKey, _ *authmap.AuthInfo) {
+			if key == nil {
+				return
+			}
+			err := authMap.Delete(*key)
```
Comment (mhofstetter) [inline cilium/cmd/bpf_auth_flush.go:42]: Re-authentication checks for existing (and valid) entries in the auth map cache. Therefore, deleting the entries from the auth map alone doesn't trigger a re-authentication and re-population of the auth map entry on subsequent requests.

https://github.com/cilium/cilium/blob/main/pkg/auth/manager.go#L127-L137

It is also visible from the log messages

```
level=debug msg="Handle authentication request" key="localIdentity=62189, remoteIdentity=64335, remoteNodeID=0, authType=spire" subsys=auth
level=debug msg="Already authenticated, skipping authentication" key="localIdentity=62189, remoteIdentity=64335, remoteNodeID=0, authType=spire" subsys=auth
level=debug msg="Clearing pending authentication" key="localIdentity=62189, remoteIdentity=64335, remoteNodeID=0, authType=spire" subsys=auth
```

Currently, we only read from the bpf auth map once at startup to populate the auth map cache.

-> To properly implement flushing the auth bpf map, the CLI would need to instruct the agent to flush the map AND the cache. (or push down the auth information in this case where an auth request for an already authenticated packet gets emitted)

**Review by mhofstetter — CHANGES_REQUESTED** [review]

File: cilium/cmd/bpf_auth_flush.go (hunk 2 above)
Comment (joestringer) [reply cilium/cmd/bpf_auth_flush.go:42]: Heads up that any `cilium bpf ...` commands should only talk directly to the bpf datapath, and not rely on the agent. This is to ensure that these commands can still operate even if the agent is unhealthy. If you want to properly flush the userspace cache as well, I suggest moving this to `cilium auth flush` or similar.

Comment (meyskens) [reply cilium/cmd/bpf_auth_flush.go:42]: > Re-authentication checks for existing (and valid) entries in the auth map cache. Therefore, deleting the entries from the auth map alone doesn't trigger a re-authentication and re-population of the auth map entry on subsequent requests.

This is why i wrote this to test a way to fix that, we should not rely on the cache always being correct as external things can delete the bpf entries. There is a 2nd PR open to fix that.

see https://github.com/cilium/cilium/issues/27183

**Review by sayboras — APPROVED** [review]
LGTM :heavy_check_mark:

File: cilium/cmd/bpf_auth_flush.go (hunk 2 above)
Comment (mhofstetter) [reply cilium/cmd/bpf_auth_flush.go:42]: ok :) would be great to have such information in the PR description itself :)

**Review by mhofstetter — APPROVED** [review]

**Pull request merged**